### ISSUE TITLE
Support the macro language inside enums

### ISSUE THREAD

**Issue by ysbaddaden** [crystal-lang/crystal#2421]
The following fails to compile with `Syntax error in eval:4: unexpected token: ifdef`:

```crystal
enum Something
  VAL = 1

  ifdef darwin
    VAR = 2
  end
end
```

See https://play.crystal-lang.org/#/r/w5j

**Comment by asterite** [comment]
I can try to implement this. And maybe try to implement `ifdef` inside struct/union definitions (that doesn't work either).

**Comment by luislavena** [comment]
That will be excellent @asterite and will reduce a bit of duplication when defining structures/enums across POSIX platforms (I know, right now `linux`/`darwin`, but there are others) :smile:

**Comment by keplersj** [comment]
Is this still valid with #2710?

**Comment by jhass** [comment]
Kind of, we still can't have macros in enum's

```cr
enum Something
  VAL = 1

  {% if flag?(:darwin) %}
    VAR = 2
  {% end %}
  end
end
```
```
Syntax error in ./bug.cr:4: unexpected token: {%

  {% if flag?(:darwin) %}
```

Though I'm not sure whether that perhaps makes it a duplicate of #424

**Comment by keplersj** [comment]
I think that this issue is a dupe now.

**Comment by jhass** [comment]
well #424 is about macro calls.

**Comment by keplersj** [comment]
Okay, I see the difference now.

**Comment by asterite** [comment]
I can probably fix this for 0.18.0

**Comment by ysbaddaden** [comment]
Since it was impossible, not implementing ifdef in enum is fine, since it would just end up being removed in favor to macro calls with `flags?` a bit later.
